### ISSUE TITLE
If you cancel a job, the sauceconnect tunnel stays around

### ISSUE THREAD

**Issue by d3ming** [circleci/sauce-connect#8]
Is there a way to force the `- killall --wait sc || true` command to run at the end of a job?  I'm finding that I'm leaking SC tunnels when jobs get cancelled so it never gets to the kill command after a tunnel was started

**Comment by bellkev** [comment]
I'm afraid we don't have a step that runs post-cancellation. When I last played with this, it seemed like Sauce does "garbage collect" unused tunnels after a few minutes. Do you know if that is in fact the case or do they stay leaked indefinitely?

**Comment by d3ming** [comment]
They leak for at least hours, not indefinitely I don't think but I had to close tunnels manually several times because we run out.

I noted this to CircleCI support as well but I think it would definitely be good to have a stage in the circle.yml file for `teardown` tasks that always run at the end

**Comment by bellkev** [comment]
Yeah, that's definitely a reasonable request. I'll make sure it's on the list internally.
